core/modules/wcontrol/TransmitMgr.h

```diff
@@ -29,15 +29,55 @@
 #include <atomic>
 #include <condition_variable>
 #include <mutex>
+#include <map>
 
 // Qserv headers
+#include "global/intTypes.h"
 
 
 namespace lsst {
 namespace qserv {
 namespace wcontrol {
 
 class TransmitLock;
+class QidMgr;
+
+class LockCount {
+public:
+    LockCount() = default;
+
+    friend QidMgr;
+private:
+    int _take(); /// @return _totalCount.
+    int _release(); /// @return _totalCount.
+
+    std::atomic<int> _totalCount{0};
+    std::atomic<int> _count{0};
+    std::atomic<int> _maxCount{1};
```

Comment:
What's a reason for using the signed integer type for counters?

Reply:
I'm doing subtraction with these. Unsigned are inherently dangerous in subtraction and the int range is far more than I would ever expect to need for this purpose.